src/Stratis.Bitcoin.Features.Wallet/Broadcasting/FullNodeBroadcasterManager.cs

```diff
@@ -13,13 +13,9 @@ public class FullNodeBroadcasterManager : BroadcasterManagerBase
         /// <summary>Memory pool validator for validating transactions.</summary>
         private readonly IMempoolValidator mempoolValidator;
 
-        /// <summary>Connection manager for managing node connections.</summary>
-        private readonly IConnectionManager connectionManager;
-
-        public FullNodeBroadcasterManager(IConnectionManager connectionManager, IMempoolValidator mempoolValidator) : base()
+        public FullNodeBroadcasterManager(IConnectionManager connectionManager, IMempoolValidator mempoolValidator) : base(connectionManager)
         {
-            this.connectionManager = connectionManager ?? throw new ArgumentNullException(nameof(connectionManager));
-            this.mempoolValidator = mempoolValidator ?? throw new ArgumentNullException(nameof(mempoolValidator));
+            this.mempoolValidator = mempoolValidator;
```

Comment:
You removed the check to check that `mempoolValidator` is not null.
Please reinstate this check by using `Guard.NotNull` instead.

Reply:
Ok. 
The reason why I did that is because we almost never have those on constructors that get created through DI. 
But I guess since it's supposed to be a refactoring the check should be there.